app/src/test/java/cz/quanti/razym/rocketapp/presentation/RocketDetailModelTest.kt

```diff
@@ -0,0 +1,139 @@
+    private val repository = createRepository()
+
+    @Before
+    fun setup() {
+        Dispatchers.setMain(testDispatcher)
+    }
+
+    @After
+    fun cleanup() {
+        Dispatchers.resetMain()
+    }
+
+    @Test
+    fun `should convert data to model`() {
+        val data = rocketsData.first { it.id == ids[0] }
+
+        val model = data.asRocketDetail()
+
+        model.id shouldBe ids[0]
+        model.name shouldBe "Falcon 1"
+        model.firstFlight shouldBe "2006-03-24"
+        model.overview shouldStartWith "The Falcon 1 was an"
+        model.heightMeters shouldBe 22.25
+        model.diameterMeters shouldBe 1.68
+        model.massKg shouldBe 30146.0
+        model.firstStage.burnTimeSec shouldBe 169
+        model.firstStage.engines shouldBe 1
+        model.firstStage.fuelAmountTons shouldBe 44.3
+        model.firstStage.reusable shouldBe false
+        model.secondStage.burnTimeSec shouldBe 378
+        model.secondStage.engines shouldBe 1
+        model.secondStage.fuelAmountTons shouldBe 3.38
+        model.secondStage.reusable shouldBe false
+        model.flickrImages.size shouldBe 2
+    }
+
+    @Test
+    fun `uiState should be loading first`() = runTest(testDispatcher) {
+        // repository is mocked
+        val id = ids[0]
+
+        val viewModel = RocketDetailViewModel(repository)
+        viewModel.fetchRocket(id)
+
+        // Loading state until we let the coroutine in model work with advanceUntilIdle()
+        viewModel.uiState.value.state shouldBe UiState.Loading
+
+        advanceUntilIdle()
+        viewModel.uiState.value.state shouldNotBe UiState.Loading
+    }
+
+    @Test
+    fun `uiState should be success`() = runTest(testDispatcher) {
+        // repository is mocked
+        val id = ids[0]
+
+        val viewModel = RocketDetailViewModel(repository)
+        viewModel.fetchRocket(id)
+
+        advanceUntilIdle()
+        assertSuccess(viewModel, id)
+    }
+
+    @Test
+    fun `uiState should be error`() = runTest(testDispatcher) {
+        // repository is mocked
+        val id = "non-existing-id"
+
+        val viewModel = RocketDetailViewModel(repository)
+        viewModel.fetchRocket(id)
+
+        advanceUntilIdle()
+        val error = viewModel.uiState.value.state as UiState.Error
+        error shouldNotBe null
+    }
+
+    @Test
+    fun `uiState should be updated after fetch`() = runTest(testDispatcher) {
+        // repository is mocked
+
+        val viewModel = RocketDetailViewModel(repository)
+
+        for (id in ids) {
+            viewModel.fetchRocket(id)
+            advanceUntilIdle()
+            assertSuccess(viewModel, id)
+        }
+    }
+
+    private fun createRepository() : RocketsRepository {
+        val idSlot = slot<String>()
+        return mockk {
+            coEvery { getRocket(capture(idSlot)) } returns
+                    flow { emit(rocketsData.first { it.id == idSlot.captured }) }
+        }
+    }
```

Comment:
v testech chceme mit co nejjednodussi logiku (idealne zadne for cykly, ify atd.)
```suggestion
        viewModel.fetchRocket(ids[0])
        advanceUntilIdle()
        assertSuccess(viewModel, rocketsData[0].id)

        viewModel.fetchRocket(ids[1])
        advanceUntilIdle()
        assertSuccess(viewModel, rocketsData[1].id)
    }

    private fun createRepository() : RocketsRepository = mockk {
        coEvery { getRocket(ids[0]) } returns flowOf(rocketsData[0])
        coEvery { getRocket(ids[1]) } returns flowOf(rocketsData[1])
        coEvery { getRocket(errorId) } returns flow { throw Exception() }
    }
```
errorId je treba zadefinovat nekde nahore jak mas ten ids list